src/tribler-core/tribler_core/restapi/tests/test_debug_endpoint.py

```diff
@@ -82,61 +84,124 @@ async def test_dump_memory(enable_api, session):
+    :param process: Either 'core' or 'gui'
+    :param log_message: log line to write to file.
+    :param num_logs: Number of log lines to write
+    :return: None
+    """
     if not log_dir.exists():
         os.makedirs(log_dir)
 
-    # Fill logging files with statements
-    core_info_log_file_path = log_dir / 'tribler-core-info.log'
+    info_log_file_path = log_dir / f'tribler-{process}-info.log'
+    log_message = log_message if log_message else f"This is a {process} test log message."
+
+    with open(info_log_file_path, "w") as info_log_file:
+        for log_index in range(num_logs):
+            info_log_file.write(f"{log_message} {log_index}\n")
 
-    # write 100 test lines which is used to test for its presence in the response
-    with open(core_info_log_file_path, "w") as core_info_log_file:
-        for log_index in range(max_lines):
-            core_info_log_file.write("%s %d\n" % (test_core_log_message, log_index))
 
-    json_response = await do_request(session, 'debug/log?process=core&max_lines=%d' % max_lines, expected_code=200)
+@pytest.mark.asyncio
+async def test_debug_pane_core_logs(enable_api, session):
+    """
+    Test whether the API returns the logs
+    """
+    log_dir = session.config.get_log_dir()
+    process = 'core'
+    test_core_log_message = "This is the core test log message"
+    num_logs = 100
+
+    create_dummy_logs(log_dir, process='core', log_message=test_core_log_message, num_logs=num_logs)
+
+    json_response = await do_request(session, f'debug/log?process={process}&max_lines={num_logs}', expected_code=200)
     logs = json_response['content'].strip().split("\n")
 
     # Check number of logs returned is correct
-    assert len(logs) == max_lines
+    assert len(logs) == num_logs
 
     # Check if test log message is present in the logs, at least once
     log_exists = any((True for log in logs if test_core_log_message in log))
     assert log_exists, "Test log not found in the debug log response"
 
 
+@pytest.mark.asyncio
+async def test_debug_pane_core_logs_in_root_dir(tribler_root_dir, enable_api, session):
+    """
+    Test whether the API returns the logs when logs are present in the root directory.
+    """
+
+    # Tribler logs are by default set to root state directory. Here we define the
+    # root state directory by updating 'TSTATEDIR' environment variable.
+    old_env_state_dir = os.environ.get('TSTATEDIR', None)
+    reset_env_state_directory(str(tribler_root_dir))
+
+    log_dir = get_root_state_directory()
+
+    process = 'core'
+    num_logs = 100
+
+    create_dummy_logs(log_dir, process='core', num_logs=num_logs)
+
+    json_response = await do_request(session, f'debug/log?process={process}&max_lines={num_logs}', expected_code=200)
+    logs = json_response['content'].strip().split("\n")
+
+    # Check number of logs returned is correct
+    assert len(logs) == num_logs
+
+    reset_env_state_directory(old_env_state_dir)
```

Comment:
And if the test fails, will this line be executed? The correct and safe way to do this is to do it through pytest fixtures teardown (e.g. save the var state in the fixture startup and restore it after `yield` in the fixture).

Also, I don't understand why you really need to use `get_root_state_directory()` here. The intention of the test is not to test `get_root_state_directory()`, right? You're not testing the real chain of calls `run_tribler->get_root_...->session->logger` either. Then why bother? Just use a fake tmp dir.

Reply:
Good point. I have updated the PR with fixtures.

Regarding the intention of the test, it is to test that log files in the root directory is taken as fallback to display the logs in the debug pane. This is because the logs are actually written in the root directory as per the current implementation (https://github.com/Tribler/tribler/blob/devel/src/run_tribler.py) but the tests were testing the preset state directory.